addons/account/populate/res_company.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+import logging
+
+from odoo import models
+import random
+from functools import lru_cache
+
+
+_logger = logging.getLogger(__name__)
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    def _populate(self, size):
+        @lru_cache()
+        def search_coa_ids(currency_id):
+            return self.env['account.chart.template'].search([('currency_id', '=', currency_id)])
+
+        records = super()._populate(size)
+        _logger.info('Loading Chart Template')
+        default_chart_templates = self.env['account.chart.template'].search([], limit=1)
+        if not default_chart_templates:
+            # TODO install l10n_generic_coa ?
+            pass
+        for company in records[:3]:
+            chart_templates_cur = search_coa_ids(company.currency_id.id) or default_chart_templates
```

Comment:
The duplicate CHF was a mistake. About the order, I don't care you can change it. Anyway, this was a quick draft, as wan said you may want to test active currencies, not only those one. I think that we should chose a way to go on this point and stick to it. SO maybe this should be adapt to a search on active currencies (with an arbitrary limit and a custom order maybe)

Reply:
If I'm not mistaken, the `base.main_company` is untouched by the populate, so the order of currencies doesn't really matter. I added a commit to remove the CHF duplicate though.

Also the CHF currency is activated automatically by the create of `res.company` in the `base` module so there is not really an issue here.

I added a commit to use iteratively all the active currencies instead of those in a list. It will be only USD and EUR on a clean DB, but it is easy to change if we want something else for some use case.